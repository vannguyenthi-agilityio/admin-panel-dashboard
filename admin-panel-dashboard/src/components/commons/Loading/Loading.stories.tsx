import type { Meta, StoryObj } from "@storybook/react";

// Components
import Loading from ".";

const meta = {
  title: "Components/Commons/Loading",
  component: Loading,
  tags: ["autodocs"],
  argTypes: {
    width: { description: "Width class of loading" },
    height: { description: "Height class of loading" },
    fillColor: { description: "Color of loading" },
  },
} as Meta<typeof Loading>;

export default meta;

type Story = StoryObj<typeof meta>;

export const LoadingDefault: Story = {
  args: {},
};

export const LoadingCustomSize: Story = {
  args: {
    width: "h-[16px]",
    height: "h-[16px]",
    fillColor: "#241212",
  },
};
