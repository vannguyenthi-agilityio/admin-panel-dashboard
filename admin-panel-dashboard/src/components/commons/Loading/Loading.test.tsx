import { render } from "@testing-library/react";

// Components
import Loading from ".";

describe("Loading component", () => {
  it("Should render Loading snapshot correctly", () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });

  it("Should render Loading snapshot full width", () => {
    const { container } = render(<Loading isFullWidth={true} />);
    expect(container).toMatchSnapshot();
  });
});
