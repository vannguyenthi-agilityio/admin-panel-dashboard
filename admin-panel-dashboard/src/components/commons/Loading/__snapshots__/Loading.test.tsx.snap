// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component Should render Loading snapshot correctly 1`] = `
<div>
  <div
    role="status"
  >
    <svg
      aria-hidden="true"
      class="h-16 h-16 text-transparent animate-[spin_0.4s_linear_infinite] dark:text-gray-600 fill-black dark:fill-white"
      fill="none"
      viewBox="0 0 22 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 3C11 2.44772 11.4477 2 12 2C17.5228 2 22 6.47715 22 12C22 14.7175 20.9147 17.1835 19.1562 18.9849C18.7704 19.3801 18.1373 19.3878 17.7421 19.002C17.3469 18.6162 17.3392 17.983 17.725 17.5879C19.1336 16.1449 20 14.1746 20 12C20 7.58172 16.4183 4 12 4C11.4477 4 11 3.55228 11 3Z"
        fill="black"
      />
    </svg>
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`Loading component Should render Loading snapshot full width 1`] = `
<div>
  <div
    class="fixed bg-[rgba(0,0,0,0.5)] overflow-hidden w-full h-full inset-0 z-50 cursor-not-allowed"
  >
    <div
      class="h-full justify-center"
    >
      <div
        role="status"
      >
        <svg
          aria-hidden="true"
          class="h-16 h-16 text-transparent animate-[spin_0.4s_linear_infinite] dark:text-gray-600 fill-white"
          fill="none"
          viewBox="0 0 22 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 3C11 2.44772 11.4477 2 12 2C17.5228 2 22 6.47715 22 12C22 14.7175 20.9147 17.1835 19.1562 18.9849C18.7704 19.3801 18.1373 19.3878 17.7421 19.002C17.3469 18.6162 17.3392 17.983 17.725 17.5879C19.1336 16.1449 20 14.1746 20 12C20 7.58172 16.4183 4 12 4C11.4477 4 11 3.55228 11 3Z"
            fill="currentFill"
          />
        </svg>
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</div>
`;
